includes/admin/class-wc-stripe-admin-notices.php

```diff
@@ -92,12 +92,12 @@ public function get_payment_methods() {
 			'Alipay'     => 'WC_Gateway_Stripe_Alipay',
 			'Bancontact' => 'WC_Gateway_Stripe_Bancontact',
 			'EPS'        => 'WC_Gateway_Stripe_EPS',
-			'Giropay'    => 'WC_Gateway_Stripe_Giropay',
-			'iDeal'      => 'WC_Gateway_Stripe_Ideal',
+			'giropay'    => 'WC_Gateway_Stripe_Giropay',
+			'iDEAL'      => 'WC_Gateway_Stripe_Ideal',
```

Comment:
Good call, on reducing confusion here. The functionality to show the notices does rely on lowercased payment method names.

But notice that on line [254](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/494d091ca1b0420883c0ea272b6a26954ee8ae30/includes/admin/class-wc-stripe-admin-notices.php#L254), `payment_methods_check_environment()`, also uses the array `key` to display the message in the notice, like the following image demonstrates:

![image](https://user-images.githubusercontent.com/6844516/144658883-818d2592-40c9-4c87-a7e2-d28699ef1f0b.png)

> **Note**: Line 254 would get called when one of the payment methods/gateways is enabled but not through the <em>new Stripe checkout experience</em>.

If we lowercase the actual array `keys`, we would get something like the following, instead:
![image](https://user-images.githubusercontent.com/6844516/145057880-794c2e61-e650-4a7b-b6bd-9b56aca5afc0.png)

I do agree that it's a bit odd that we're using the array keys to display the payment method names in the notice but for now, we can leave it as is. Unless we want to go ahead and refactor things so that the payment method classes include the payment method name - like the UPE payment method classes do - and we can just retrieve the name from the class itself.

Thoughts?

Reply:
Refactoring is a good idea, but it's out of the scope of this PR. We can merge this PR and open a new issue for this.

Getting the method name from the class seems to be the way to go.